### PR TITLE
deploy: set condition reason correctly for new RCs

### PR DESCRIPTION
**Pull request by kargakis**
@mfojtik oversight, ptal

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

File: test/extended/deployments/deployments.go [hunk 1]
```diff
@@ -782,4 +784,44 @@ var _ = g.Describe("deploymentconfigs", func() {
...
+					return false, nil
+				}
+				conditions = dc.Status.Conditions
+				cond := deployutil.GetDeploymentCondition(dc.Status, deployapi.DeploymentProgressing)
+				return cond != nil && cond.Reason == deployutil.NewReplicationControllerReason, nil
+			})
+			if err == wait.ErrWaitTimeout {
+				err = fmt.Errorf("deployment config %q never updated its conditions: %#v", name, conditions)
+			}
+			o.Expect(err).NotTo(o.HaveOccurred())
```
Comment (mfojtik) [inline test/extended/deployments/deployments.go]: as some point (and we are getting to that point now) we should split this file :-)

File: test/extended/testdata/deployments/deployment-ignores-deployer.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    spec:
+      containers:
+      - env:
+        - name: MYSQL_USER
+          value: user1MM
+        - name: MYSQL_PASSWORD
+          value: wvXqW5E0
+        - name: MYSQL_DATABASE
+          value: root
+        image: openshift/mysql-55-centos7:latest
```
Comment (mfojtik) [inline test/extended/testdata/deployments/deployment-ignores-deployer.yaml]: use openshift/origin here (it does not matter for this test, but we don't want somebody to copy&paste this in a test that will actually pull this.

**Comment by mfojtik** [comment]
LGTM (but fix the image in fixture before merging)

**Comment by kargakis** [comment]
@mfojtik fixture fixed, ptal

**Comment by mfojtik** [comment]
[merge]

**Comment by kargakis** [comment]
flaked on https://github.com/openshift/origin/issues/11616

**Pull request merged**